homeassistant/components/statistics/sensor.py

```diff
@@ -277,7 +295,7 @@ def _add_state_to_queue(self, new_state):
 
         try:
             if self.is_binary:
-                self.states.append(new_state.state)
+                self.states.append(new_state.state == "on")
```

Comment:
Bigger change: Replaced binary string representation "on"/ "off" by boolean (None, Unknown etc are already filtered out). This increases performance I guess but also enabled to avoid many mypy issues as many float functions are not defined for str but for boolean...